src/backend/utils/misc/queryenvironment.c

```diff
@@ -1572,6 +1578,45 @@ ENRDropTempTables(QueryEnvironment *queryEnv)
 		object.objectSubId = 0;
 		object.objectId = enr->md.reliddesc;
 		add_exact_object_address(&object, objects);
+		/*
+		 * Delete the physical storage for the relation.
+		 * See: smgrDoPendingDeletes()
+		 */
+		rel = relation_open(enr->md.reliddesc, AccessExclusiveLock);
+		srel = smgropen(rel->rd_locator, rel->rd_backend);
+
+		/* allocate the initial array, or extend it, if needed */
+		if (maxrels == 0)
```

Comment:
why are we doing this? we have single relationship, we should just pass pointer to that relation with count as 1 to smgrdounlinkall

Reply:
We do not necessarily only have one relation - there can be multiple table variables and temp tables defined in the same scope, which would all be cleaned up at once by this function.

Reply:
Correction: currently table variables are being cleaned up prior to this function during pltsql_clean_table_variables(), which manually executes a DROP TABLE <tblname> statement to drop the table variable. Since that function only registers the physical file for deletion at transaction rather than immediately, we probably should remove that function and let this function do both table variable and temp table cleanup, but that can be done in a later commit.

TLDR: currently only temp tables in scope are being cleaned up here, and not table variables, which are handled separately.